storage-s3-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -47,6 +46,47 @@
 public abstract class BaseExternalLogStore extends HadoopFileSystemLogStore {
     private static final Logger LOG = LoggerFactory.getLogger(BaseExternalLogStore.class);
 
+    /**
+     * The delay, in seconds, after an external entry has been committed to the delta log at which
+     * point it is safe to be deleted from the external store.
+     *
+     * We want a delay long enough such that, after the external entry has been deleted, another
+     * write attempt for the SAME delta log commit can FAIL using ONLY the FileSystem's existence
+     * check (e.g. `fs.exists(path)`). Recall we assume that the FileSystem does not provide mutual
+     * exclusion.
+     *
+     * We use a value of 1 day.
+     *
+     * If we choose too small of a value, like 0 seconds, then the following scenario is possible:
+     * - t0:  Writers W1 and W2 start writing data files
+     * - t1:  W1 begins to try and write into the _delta_log.
+     * - t2:  W1 checks if N.json exists in FileSystem. It doesn't.
+     * - t3:  W1 writes actions into temp file T1(N)
+     * - t4:  W1 writes to external store entry E1(N, complete=false)
+     * - t5:  W1 copies (with overwrite=false) T1(N) into N.json.
+     * - t6:  W1 overwrites entry in external store E1(N, complete=true, expireTime=now+0)
+     * - t7:  E1 is safe to be deleted, and some external store TTL mechanism deletes E1
+     * - t8:  W2 begins to try and write into the _delta_log.
+     * - t9:  W1 checks if N.json exists in FileSystem, but too little time has transpired between
```

Comment:
Yes, exactly.

Reply:
@scottsand-db Trying to verify my understanding of this comment, does this mean TTL can be set to 0 if using S3 as storage due to the fact that it's strongly consistent?

Reply:
I wouldn't encourage that.

S3 doesn't provide mutual exclusion on writes. We need to ensure that two concurrent writers have some sufficient delta time between the two writes to avoid any mutual exclusion issues. If you set the TTL to 0 ... then you are just getting closer and closer to a mutual exclusion problem, with what benefit?